e2etests/pkg/frr/bgp.go

```diff
@@ -130,3 +155,25 @@ func parseNeighbour(vtyshRes string) (*FRRNeighbor, error) {
 	}
 	return nil, nil
 }
+
+func parseRoutes(vtyshRes string) (BGPRoutes, error) {
+	res := BGPRoutes{}
+	parsed := IPInfo{}
+	err := json.Unmarshal([]byte(vtyshRes), &parsed)
+	if err != nil {
+		return nil, errors.Join(err, fmt.Errorf("failed to parse vtysh response: %s", vtyshRes))
+	}
+
+	for p, routes := range parsed.Routes {
+		for _, r := range routes {
+			if !r.BestPath || !r.Valid {
+				continue
+			}
+			if len(r.Nexthops) == 0 {
+				continue
+			}
+			res[p] = r.Nexthops[0].IP
```

Comment:
why this approach instead of parsing all nexthops + going over all of them in `HaveRoute`? (if it has something to do with BestPath/Valid, can we make this more explicit?)

Reply:
will fix it.